### PR TITLE
Change PPS geometry ES producer to read preprocessed geometry from DB

### PR DESCRIPTION
**Pull request by wpcarvalho**
#### PR description:

The goal of this PR is to make the ES producer `CTPPSGeometryESModule.cc` able to read the PPS reco geometry from conditions data stored in the database. Currently, PPS geometry is read directly from XML files.

- No changes are expected in the output.
- We plan a future PR that will change the default reco sequence to consume conditions data from DB. 

@fabferro @jan-kaspar 

#### PR validation:

The following validation tests have been performed:

-  Comparison of pot and sensor positions obtained from XML (current producer) and from DB (changed producer). Perfect agreement is found.
-  Reco workflow. Local tests have been run to compare the current and the changed versions of the producer. Nearly perfect agreement has been observed, as can be seen from a sample of plots shown below. Only a few very tiny differences (probably at the round level) among hundreds of histograms. See the Mean for theta_x and theta_y histograms in the bottom canvas (top row = baseline ; bottom row = changed producer) as the only visible differences.

reco_cmp

base_vs_db_1

The tests with the changed producer used conditions data from a local sqlite file and also the same data uploaded to the Prep database as a test tag: https://cms-conddb.cern.ch/cmsDbBrowser/list/Prep/tags/PPS_RecoGeometry_test_v1.

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Not a backport.

### CODE REVIEW

**Comment by civanch** [comment]
please test

File: Geometry/VeryForwardGeometryBuilder/src/DetGeomDesc.cc [hunk 1]
```diff
@@ -75,6 +75,38 @@ DetGeomDesc::DetGeomDesc(const DetGeomDesc& ref, CopyMode cm) {
   m_z = ref.m_z;
 }
 
+// Constructor from DB object PDetGeomDesc::Item
+DetGeomDesc::DetGeomDesc(const PDetGeomDesc::Item& item)
+    : m_name(item.name_),
+      m_copy(item.copy_),
+      m_isDD4hep(true),
+      m_params(item.params_),  // default unit from DD4hep (cm)
```
Comment (cvuosalo) [inline Geometry/VeryForwardGeometryBuilder/src/DetGeomDesc.cc:83]:
```suggestion
      m_params(item.params_),  // default unit from DD4hep 
```

File: Geometry/VeryForwardGeometryBuilder/src/DetGeomDesc.cc [hunk 2]
```diff
@@ -75,6 +75,38 @@ DetGeomDesc::DetGeomDesc(const DetGeomDesc& ref, CopyMode cm) {
...
+      m_sensorType(item.sensorType_),
+      m_geographicalID(item.geographicalID_),
+      m_z(item.z_)  // converted from cm (DD4hep) to mm
```
Comment (cvuosalo) [inline Geometry/VeryForwardGeometryBuilder/src/DetGeomDesc.cc:86]:
```suggestion
      m_z(item.z_)  // converted from DD4hep units to mm
```

File: Geometry/VeryForwardGeometryBuilder/src/DetGeomDesc.cc (hunk 1 above)
Comment (wpcarvalho) [reply Geometry/VeryForwardGeometryBuilder/src/DetGeomDesc.cc]: Done.

File: Geometry/VeryForwardGeometryBuilder/src/DetGeomDesc.cc (hunk 2 above)
Comment (wpcarvalho) [reply Geometry/VeryForwardGeometryBuilder/src/DetGeomDesc.cc]: Done.

**Comment by cvuosalo** [comment]
+1

**Comment by cvuosalo** [comment]
@cmsbuild please test

**Comment by qliphy** [comment]
+1

**Pull request merged**